packages/gatsby/src/commands/develop.ts

```diff
@@ -1,631 +1,240 @@
+  } catch (e) {
+    return null
+  }
 
-async function startServer(program: IDevelopArgs): Promise<IServer> {
-  const indexHTMLActivity = report.phantomActivity(`building index.html`, {})
-  indexHTMLActivity.start()
-  const directory = program.directory
-  const directoryPath = withBasePath(directory)
-  const workerPool = WorkerPool.create()
-  const createIndexHtml = async (activity: ActivityTracker): Promise<void> => {
-    try {
-      await buildHTML({
-        program,
-        stage: Stage.DevelopHTML,
-        pagePaths: [`/`],
-        workerPool,
-        activity,
-      })
-    } catch (err) {
-      if (err.name !== `WebpackError`) {
-        report.panic(err)
-        return
-      }
-      report.panic(
-        report.stripIndent`
-          There was an error compiling the html.js component for the development server.
+  try {
+    return require(file)
+  } catch (e) {
+    return null
+  }
+}
 
-          See our docs page on debugging HTML builds for help https://gatsby.dev/debug-html
-        `,
-        err
-      )
+// Heuristics for gatsby-config.js, as not all changes to it require a full restart to take effect
+const doesConfigChangeRequireRestart = (
+  lastConfig: Record<string, any>,
+  newConfig: Record<string, any>
+): boolean => {
+  // Ignore changes to siteMetadata
+  const replacer = (_, v): string | void => {
+    if (typeof v === `function`) {
+      return v.toString()
+    } else {
+      return v
     }
   }
 
-  await createIndexHtml(indexHTMLActivity)
-
-  indexHTMLActivity.end()
-
-  // report.stateUpdate(`webpack`, `IN_PROGRESS`)
-
-  const webpackActivity = report.activityTimer(`Building development bundle`, {
-    id: `webpack-develop`,
-  })
-  webpackActivity.start()
-
-  const devConfig = await webpackConfig(
-    program,
-    directory,
-    `develop`,
-    program.port,
-    { parentSpan: webpackActivity.span }
+  const oldConfigString = JSON.stringify(
+    { ...lastConfig, siteMetadata: null },
+    replacer
   )
-
-  const compiler = webpack(devConfig)
-
-  /**
-   * Set up the express app.
-   **/
-  const app = express()
-  app.use(telemetry.expressMiddleware(`DEVELOP`))
-  app.use(
-    webpackHotMiddleware(compiler, {
-      log: false,
-      path: `/__webpack_hmr`,
-      heartbeat: 10 * 1000,
-    })
+  const newConfigString = JSON.stringify(
+    { ...newConfig, siteMetadata: null },
+    replacer
   )
 
-  app.use(cors())
+  if (oldConfigString === newConfigString) return false
 
-  /**
-   * Pattern matching all endpoints with graphql or graphiql with 1 or more leading underscores
-   */
-  const graphqlEndpoint = `/_+graphi?ql`
+  return true
+}
 
-  if (process.env.GATSBY_GRAPHQL_IDE === `playground`) {
-    app.get(
-      graphqlEndpoint,
-      graphqlPlayground({
-        endpoint: `/___graphql`,
-      }),
-      () => {}
-    )
-  } else {
-    graphiqlExplorer(app, {
-      graphqlEndpoint,
-    })
+class ControllableScript {
```

Comment:
Do we want to keep this class here? or move it somewhere else? (like utils)

Reply:
As long as we only use it here I'd vote for only using it here? I don't think we want to guarantee reusability at this point.

Reply:
For me it just clutters the file having an inline class, that's all but the reason why to leave it here is valid